web/src/test/scala/quasar/api/services/DataServiceSpec.scala

```diff
@@ -0,0 +1,511 @@
+        todo // Figure out what we are doing with nested backends because not sure this test makes sense without them
+      }
+      "be 201 with file" ! prop {(source: Path[Abs,File,Sandboxed], destination: Path[Abs,File,Sandboxed], data: Vector[Data]) =>
+        val sourceString = posixCodec.printPath(source)
+        val destinationString = posixCodec.printPath(destination)
+        val request = Request(uri = Uri(path = sourceString), headers = Headers(Header("Destination", destinationString)), method = Method.MOVE)
+        val filesystem = InMemState.fromFiles(Map(source -> data))
+        val response = service(filesystem)(request).run
+        response.status must_== Status.Created
+        // TODO: Check that the file was moved properly (not in old test though...)
+      }
+      "be 201 with dir" ! prop {(sourceDir: Path[Abs,Dir,Sandboxed], sampleFiles: Map[String, Vector[Data]], destinationDir: Path[Abs,Dir,Sandboxed]) =>
+        val sourceDirString = posixCodec.printPath(sourceDir)
+        val fileMapping = sampleFiles.map{ case (fileName,data) => (sourceDir </> file(fileName), data)}
+        val destinationString = posixCodec.printPath(destinationDir)
+        val request = Request(uri = Uri(path = sourceDirString), headers = Headers(Header("Destination", destinationString)), method = Method.MOVE)
+        val filesystem = InMemState fromFiles fileMapping
+        val response = service(filesystem)(request).run
+        response.status must_== Status.Created
+        // TODO: Check that the file was moved properly (not in old test though...)
+      }
+      "be 501 for src and dst not in the same backend" >> todo // What are we doing? make sure that we test all error conditions that the backend can return
+    }
+    "DELETE" >> {
+      "be 404 for missing backend" >> todo // What are we doing?
+      "be 200 with existing file" ! prop { (file: Path[Abs,File,Sandboxed], data: Vector[Data]) =>
+        val path = posixCodec.printPath(file)
+        val request = Request(uri = Uri(path = path), method = Method.DELETE)
+        val filesystem = InMemState fromFiles (Map(file -> data))
+        val response = service(filesystem)(request).run
+        response.status must_== Status.Ok
+        // Check if file was actually deleted
+      }
+      "be 200 with existing dir" ! prop { (dir: Path[Abs,Dir,Sandboxed], sampleFiles: Map[String, Vector[Data]]) =>
+        val dirPath = posixCodec.printPath(dir)
+        val fileMapping = sampleFiles.map{ case (fileName,data) => (dir </> file(fileName), data)}
+        val request = Request(uri = Uri(path = dirPath), method = Method.DELETE)
+        val filesystem = InMemState fromFiles fileMapping
+        val response = service(filesystem)(request).run
+        response.status must_== Status.Ok
+        // Check if dir was actually deleted
+      }
+      "be 200 with missing file (idempotency)" ! prop { (file: Path[Abs,File,Sandboxed], data: Vector[Data]) =>
```

Comment:
@wemrysi and I talked about this briefly the other day and I believe he's right; as long as the effect is the same as a successful delete (the file doesn't exist afterward), we *can* return 404. It's not obvious to me which is the right way to go and it would be nice to have a use case coming from the front-end, but in the absence of that I'm OK with the change.